Core/Src/control.c

```diff
@@ -89,39 +57,28 @@ void startControlTask(){
 
  */
 
-void OverflowCheck(TIM_HandleTypeDef * htim){
 
-	//instance for tim2
-	if(htim -> Instance == TIM2){
-		TIM2_OVC++;
-		if(TIM2_OVC >= 2){
-			TIM2_freq = 0;
-		}
-	}
-	//instance for tim3 (channel checks)
-	if (htim -> Instance == TIM3){
-		if(htim -> Channel == HAL_TIM_ACTIVE_CHANNEL_1){
-			TIM3_CH1_OVC ++;
-			if(TIM3_CH1_OVC >= 2){
-				TIM3_CH1_freq = 0;
-			}
-		}
-		if(htim -> Channel == HAL_TIM_ACTIVE_CHANNEL_2){
-			TIM3_CH2_OVC ++;
-			if(TIM3_CH2_OVC >= 2){
-				TIM3_CH2_freq = 0;
-			}
-		}
-	}
-	//instance for tim4
-	if(htim -> Instance == TIM4){
-		TIM4_OVC ++;
-		if(TIM4_OVC >= 2){
-			TIM4_freq = 0;
-		}
-	}
+
+
+const TIM_HandleTypeDef* wheel_to_timer_mapping[NUM_WHEELS] = {&htim2, &htim3, &htim3, &htim4}; // Note: TIM3 is repeated for both channels CH1 and CH2
+
+// Function to manage overflow conditions for each wheel's timer
+void ManageTimerOverflow(TIM_HandleTypeDef * htim) {
```

Comment:
```suggestion
void manageWheelSpeedTimerOverflow(const TIM_HandleTypeDef * htim) {
```
Watch the style guide, and if you name it better, you don't need the comment describing what the function does.